### DISCUSSION TITLE
needs-validation-feedback

### DISCUSSION THREAD

**Discussion by ZhangChengLin** [twbs/bootstrap#30659] [General]
---
[DEMO](https://codepen.io/zhangchenglin/pen/YzyQNdp)
---

This is an adaptive form structure, so the form text prompts cannot be well laid out and need to be customized. In addition, if four elements such as valid-feedback are placed, the input element will not have rounded corners. So, where should my valid-feedback element be?

There may be other problems in the form, such as discovery, hope to deal with together.

---
1. Need to add needs-validation-feedback style
2. Under this structure, the invalid-feedback div is placed there
---

**Comment by ZhangChengLin** [comment]
I did n’t read it clearly, I do n’t know why it was posted here, so it seems that I ca n’t go back to the issue

@mdo

**Comment by mdo** [comment]
Are you suggesting a feature, reporting an issue, or asking for help on a general how-to/troubleshooting issue?

**Reply by ZhangChengLin** [reply]
I originally wanted to post a bug, but did not read it clearly before posting it in this section.
I feel this should be a bug, because in the demo, invalid-feedback is nowhere to be placed.

**Comment by mdo** [comment]
> In addition, if four elements such as valid-feedback are placed, the input element will not have rounded corners

This is a known issue, and there's no workaround for it in v4 unfortunately without a breaking change of additional classes or something else.
